### ISSUE TITLE
AddressSanitizer: heap-buffer-overflow src/olvm.c:2181 in runtime

### ISSUE THREAD

**Issue by XVilka** [yuriy-chumak/ol#18]
Happened on some scripts - interestingly appears after I remove one of the comments...
```
================================================================
==6947==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x7f1393a44c40 at pc 0x00000040937f bp 0x7ffc70be4b20 sp 0x7ffc70be4b10
WRITE of size 1 at 0x7f1393a44c40 thread T0
    #0 0x40937e in runtime src/olvm.c:2181
    #1 0x41e2c5 in OL_eval src/olvm.c:4441
    #2 0x41d70b in main src/olvm.c:4279
    #3 0x7f139d535400 in __libc_start_main (/lib64/libc.so.6+0x20400)
    #4 0x402139 in _start (/home/user/tool/ol/ol+0x402139)

0x7f1393a44c40 is located 0 bytes to the right of 17474624-byte region [0x7f139299a800,0x7f1393a44c40)
allocated by thread T0 here:
    #0 0x7f139dba6210 in realloc (/lib64/libasan.so.3+0xc7210)
    #1 0x4026cd in adjust_heap src/olvm.c:1069
    #2 0x40300b in gc src/olvm.c:1269
    #3 0x4030eb in gc src/olvm.c:1280
    #4 0x404a89 in OL__gc src/olvm.c:1553
    #5 0x405cfd in runtime src/olvm.c:1750
    #6 0x41e2c5 in OL_eval src/olvm.c:4441
    #7 0x41d70b in main src/olvm.c:4279
    #8 0x7f139d535400 in __libc_start_main (/lib64/libc.so.6+0x20400)

SUMMARY: AddressSanitizer: heap-buffer-overflow src/olvm.c:2181 in runtime
Shadow bytes around the buggy address:
  0x0fe2f2740930: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fe2f2740940: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fe2f2740950: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fe2f2740960: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fe2f2740970: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0fe2f2740980: 00 00 00 00 00 00 00 00[fa]fa fa fa fa fa fa fa
  0x0fe2f2740990: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fe2f27409a0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fe2f27409b0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fe2f27409c0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fe2f27409d0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==6947==ABORTING
```

**Comment by XVilka** [comment]
Sadly I cannot share the file caused this... Hopefully may be the callstack & line of the code will point you to the cause of error.

**Comment by yuriy-chumak** [comment]
Very interesting... There are removed code portion that checks the available stack size for vm:raw primop.
I'm returning this code and moving it under #define.

Thank you!!

**Comment by yuriy-chumak** [comment]
fixed in 8a5931d471f83bb77b4580571f43f664b5c7f8e3
